### PR TITLE
Pad buttons

### LINKED ISSUE #3: [FEATURE] Add ButtonsView
ButtonsView will be a view that can contain 1 or more buttons with configurable action (callback). It should be possible to choose either a grid or a circular layout for the buttons.

### PR DESCRIPTION
**Pull request by dudeck**
Closes #3

### CODE REVIEW

File: lib/models/pad_button_item.dart [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  /// [pressedColor] color of button when it is pressed.
+  final Color pressedColor;
+
+  /**
+   *  [supportedGestures] list of gestures for button which will call the callback [PadButtonsView.padButtonPressedCallback]
+   */
+  final List<Gestures> supportedGestures;
+
+  const PadButtonItem(
+      {this.index,
```
Comment (PabloPL) [inline lib/models/pad_button_item.dart]: Maybe we could use @required + assert to indicate what parameters are required to work?
Something like 
>const PadButtonItem({
    @required this.index,
  })  : assert(index != null);

File: lib/views/circle_view.dart [hunk 2]
```diff
@@ -10,13 +10,25 @@ class CircleView extends StatelessWidget {
...
+        this.buttonImage,
+        this.buttonText = ""});
 
   @override
   Widget build(BuildContext context) {
     return Container(
       width: size,
       height: size,
+      child:
+      Center(child: (buttonImage != null) ? buttonImage : Text(buttonText)),
```
Comment (PabloPL) [inline lib/views/circle_view.dart]: Maybe we could set that Text(buttonText) "default value" for buttonImage.
This way here we would have just
> child: Center(child: buttonImage),

File: lib/views/circle_view.dart [hunk 3]
```diff
@@ -10,13 +10,25 @@ class CircleView extends StatelessWidget {
...
+
+  final String buttonText;
+
+  CircleView(
+      {this.size,
+        this.color = Colors.transparent,
+        this.boxShadow,
+        this.border,
+        this.buttonImage,
+        this.buttonText = ""});
```
Comment (PabloPL) [inline lib/views/circle_view.dart]: Add comma here after "}" + reformat

File: lib/models/pad_button_item.dart [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      this.buttonText,
+      this.buttonImage,
+      this.backgroundColor = Colors.white54,
+      this.pressedColor = Colors.lightBlueAccent,
+      this.supportedGestures = const [Gestures.TAP]});
```
Comment (PabloPL) [inline lib/models/pad_button_item.dart]: Add comma here after "}" + reformat

File: lib/views/pad_button_view.dart [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+  final Map<int, Color> buttonsStateMap = HashMap<int, Color>();
+
+  /// [buttonsPadding] optional parameter to ad paddings for buttons.
+  final double buttonsPadding;
+
+  /// [backgroundPadButtonsColor] optional parameter, when set it shows circle.
+  final Color backgroundPadButtonsColor;
+
+  PadButtonsView({
+    this.size,
```
Comment (PabloPL) [inline lib/views/pad_button_view.dart:41]: @required + assert

File: lib/views/pad_button_view.dart [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    if (padButtonPressedCallback != null &&
+        button.supportedGestures.contains(gesture)) {
+      padButtonPressedCallback(button.index, gesture);
+      print("$gesture paddbutton id =  ${[button.index]}");
+    }
+  }
+
+  double _calculatePositionXOfButton(
+      int index, double innerCircleSize, double actualSize) {
+    double degrees = 360 / buttons.length * index;
```
Comment (PabloPL) [inline lib/views/pad_button_view.dart]: What if we pass buttons as zero-length array? Maybe we could add some assert (for example in constructor)

**Review by PabloPL — CHANGES_REQUESTED** [review]

**Review by dudeck — COMMENTED** [review]
@PabloPL updated PR after your comments. Please rereview ;)

**Review by PabloPL — APPROVED** [review]

**Review by artur-ios-dev — APPROVED** [review]

**Pull request merged**